File: RestAPI/httpPut.js
var https = require("follow-redirects").https;
var fs = require("fs");

var options = {
  method: "PUT",
  hostname: "130.195.10.173",
  port: 8080,
  path: "/api/changeowner/CAR4",
  headers: {
    "Content-Type": "application/json"
  },
  maxRedirects: 20
};

var req = https.request(options, function(res) {
  var chunks = [];

  res.on("data", function(chunk) {
    chunks.push(chunk);
  });

  res.on("end", function(chunk) {
    var body = Buffer.concat(chunks);
    console.log(body.toString());
  });

  res.on("error", function(error) {
    console.error(error);
  });
});

var postData = JSON.stringify({ owner: "XYZ" });

req.write(postData);

req.end();

File: restAPI/HTTP_PUT.go
package main

import (
"fmt"
"strings"
"net/http"
"io/ioutil"
)

func main() {

url := "130.195.10.173:8080/api/changeowner/CAR4"
method := "PUT"

payload := strings.NewReader("{\"owner\":\"XYZ\"}")

client := &http.Client {
}
req, err := http.NewRequest(method, url, payload)

if err != nil {
fmt.Println(err)
}
req.Header.Add("Content-Type", "application/json")

res, err := client.Do(req)
defer res.Body.Close()
body, err := ioutil.ReadAll(res.Body)

fmt.Println(string(body))
}
